encoding/json/decode.go

```diff
@@ -1170,6 +1171,41 @@ func (d *Decoder) decodeIntersectionType(
 
 type typeDecodingResults map[string]cadence.Type
 
+var simpleTypes = func() map[string]cadence.Type {
+	typeMap := make(map[string]cadence.Type, interpreter.PrimitiveStaticType_Count)
+
+	// Bytes is not a primitive static type
+	typeMap["Bytes"] = cadence.TheBytesType
+
+	for ty := interpreter.PrimitiveStaticType(1); ty < interpreter.PrimitiveStaticType_Count; ty++ {
+		if !ty.IsDefined() {
+			continue
+		}
+
+		cadenceType := cadence.PrimitiveType(ty)
+		if !encodeAsSimpleType(cadenceType) {
+			continue
+		}
+
+		semaType := ty.SemaType()
+
+		// Some primitive static types are deprecated,
+		// and only exist for migration purposes,
+		// so do not have an equivalent sema type
+		if semaType == nil {
+			continue
+		}
+
+		typeMap[string(semaType.ID())] = cadenceType
+	}
+
+	return typeMap
+}()
+
+func encodeAsSimpleType(primitiveType cadence.PrimitiveType) bool {
```

Comment:
The name of this function sounds like it's actually performing an encoding; can we rename it to maybe `canEncodeAsSimpleType`?

Reply:
Good idea, improved in ff83742c9a5fbe509e03e33bd262bee3fdb5b99d